src/append_only_data.rs

```diff
@@ -610,35 +625,37 @@ macro_rules! impl_appendable_data {
             /// Adds a new permissions entry.
             /// The `Perm` struct should contain valid indices.
             ///
-            /// If the specified `permissions_index` does not match the last recorded permissions
-            /// index + 1, an error will be returned.
-            fn append_permissions(&mut self, permissions: P, permissions_index: u64) -> Result<()> {
-                if permissions.entries_index() != self.entries_index() {
-                    return Err(Error::InvalidSuccessor(self.entries_index()));
+            /// If the specified `expected_index` does not equal
+            /// the count of permissions, an error will be returned.
+            fn append_permissions(&mut self, permissions: P, expected_index: u64) -> Result<()> {
+                if permissions.expected_entries_index() != self.expected_entries_index() {
+                    return Err(Error::InvalidSuccessor(self.expected_entries_index()));
                 }
-                if permissions.owners_index() != self.owners_index() {
-                    return Err(Error::InvalidOwnersSuccessor(self.owners_index()));
+                if permissions.expected_owners_index() != self.expected_owners_index() {
+                    return Err(Error::InvalidOwnersSuccessor(self.expected_owners_index()));
                 }
-                if self.permissions_index() != permissions_index {
-                    return Err(Error::InvalidSuccessor(self.permissions_index()));
+                if self.expected_permissions_index() != expected_index {
+                    return Err(Error::InvalidSuccessor(self.expected_permissions_index()));
                 }
                 self.inner.permissions.push(permissions);
                 Ok(())
             }
 
             /// Adds a new owner entry.
             ///
-            /// If the specified `owners_index` does not match the last recorded owners index + 1,
-            /// an error will be returned.
-            fn append_owner(&mut self, owner: Owner, owners_index: u64) -> Result<()> {
-                if owner.entries_index != self.entries_index() {
-                    return Err(Error::InvalidSuccessor(self.entries_index()));
+            /// If the specified `expected_index` does not equal
+            /// the count of owners index, an error will be returned.
```

Comment:
```suggestion
            /// the count of owners, an error will be returned.
```
Do you think we should change this as well?

Reply:
Absolutely, I'll do the change.